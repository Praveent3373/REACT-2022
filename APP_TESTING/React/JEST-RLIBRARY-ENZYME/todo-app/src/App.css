.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}




/* range counter styles */

.rangeCounter{
  width: 50%; margin: auto;
}
.rangeCounter_control{
  border: 2px solid tan;
  width: 250px; margin: auto;
  height: 100px; vertical-align: middle;
  line-height: 100px; border-radius: 6px;
}
.rangeCounter button{
  background: tan; color: white; border: 0; width:35px; height: 35px;
  font-size: 22px; font-weight: 500;
}
.rangeCounter span{
  padding: 0 15px; display: inline-block;
  font-size: 22px; font-weight: bold;
}
.rangeCounter_alert{
  color: red;
}
button:disabled{
  background: silver;
}