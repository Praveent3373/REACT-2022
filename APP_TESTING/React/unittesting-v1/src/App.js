import React, { useState } from 'react'
import UnitTestv1 from './components/UnitTestv1';
import UnitTest from './components/UnitTest';
export default function App() {
  return (
    <div>
      <UnitTest></UnitTest>
      <UnitTestv1></UnitTestv1>
    </div>
  )
}
