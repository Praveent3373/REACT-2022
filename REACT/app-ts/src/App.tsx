import React, { useState } from 'react';
import TsIndex from './typescript/TsIndex';
import NotesIndex from './notes/NotesIndex';

function App() {
  return (
    <>
    
      <TsIndex/>
    </>
  );
}

export default App;
