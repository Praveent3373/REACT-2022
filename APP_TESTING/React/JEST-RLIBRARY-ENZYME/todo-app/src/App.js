import logo from './logo.svg';
import './App.css';
import Todo from './components/todo/Todo';
import RangeCounterA from './components/counter/RangeCounterA'
import RangeCounterB from './components/counter/RangeCounterA'

function App() {
  return (
    <div className="App">
      {/* <Todo></Todo> */}
      {/* <RangeCounterA min={0} max={9} /> */}
      <RangeCounterB min={0} max={5} />
    </div>
  );
}

export default App;
