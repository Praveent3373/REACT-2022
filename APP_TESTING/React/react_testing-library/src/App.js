import logo from './logo.svg';
import './App.css';
import TestDomElements from './components/TestDomElements';
import TestingEvents from './components/2.TestingEvents';
import TestingAsyc from './components/3.TestingAsync';
import TestingContext from './components/4.TestingContext';
import TestRedux from './components/5.TestRedux';

function App() {
  return (
    <div className="app">
      {/* <TestDomElements></TestDomElements>
      <TestingEvents></TestingEvents> */}
      {/* <TestingAsyc></TestingAsyc> */}
      {/* <TestingContext></TestingContext> */}
      {/* <TestRedux></TestRedux> failed */}
    </div>
  );
}

export default App;
