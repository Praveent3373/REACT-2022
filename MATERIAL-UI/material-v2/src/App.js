import logo from './logo.svg';
import './App.css';
import Container from '@material-ui/core/Container';
import ComponentA from './components/ComponentA';
import ComponentB from './components/ComponentB';
import ComponentC from './components/ComponentC'
import { theme } from './theme';
import { ThemeProvider } from '@material-ui/core';
const newTheme = theme;
function App() {
  return (
    <ThemeProvider theme={newTheme}>
      <Container maxWidth ="lg">
        <ComponentA />
        <ComponentB />
        <ComponentC />
      </Container>
    </ThemeProvider>
  );
}

export default App;
