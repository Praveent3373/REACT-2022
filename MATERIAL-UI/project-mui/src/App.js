import logo from './logo.svg';
import './App.css';
import Navbar from './components/Navbar';
import { Grid, makeStyles } from '@material-ui/core';
import Leftbar from './components/Leftbar';
import Rightbar from './components/Rightbar';
import Feed from './components/Feed';
import Add from './components/Add';
import GridSample from './components/GridSample';
const useStyles = makeStyles((theme) => ({
    right: {
      [theme.breakpoints.down('sm')]: {
          display: 'none'
      }
    },
    main: {
      maxWidth: 1366,
      margin: 'auto'
    }
}))

function App() {
  const classes = useStyles();
  return (
    <div className={classes.main}>
      <Navbar></Navbar>
      <Grid container>
        <Grid item sm = {2} xs={2}><Leftbar></Leftbar></Grid>
        <Grid item sm = {7} xs={10}><Feed></Feed></Grid>
        <Grid item sm = {3} xs={2} className={classes.right}><Rightbar></Rightbar></Grid>
      </Grid>
      <Add/>
    </div>
  );
}

export default App;
