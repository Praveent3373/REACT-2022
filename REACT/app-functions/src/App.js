import logo from './logo.svg';
import './App.css';
import UseState from './hooks/01.UseState';
import UseEffect from './hooks/02.UseEffect';
import UseRef from './hooks/03.UseRef';
import UseReducer from './hooks/04.UseReducer';
import UseCustom from './hooks/05.UseCustom';
import ReactMemo from './hooks/06.ReactMemo';
import UseCallback from './hooks/07.UseCallback';
import UseMemo from './hooks/08.UseMemo';
import ClickCounter from './hoc/ClickCounter';
import HoverCounter from './hoc/HoverCounter';
import ClassCounter from './hoc/class/ClassCounter';
import FnCounter from './hoc/FnCounter';
import {IndexErrorB } from './error-boundaries/IndexErrorB';
import Counter from './hooks/09.context/Counter';
import { Lifecycle } from './lifecycle/LifecycleMethods';
import { Form } from './forms/Form';
import { MultipleInputs } from './forms/MultipleInputs';
import { MultipleReturns } from './conditional-rendering/MultipleReturns';
import { ShortCircuit } from './conditional-rendering/ShortCircuit';
import { ShowHide } from './conditional-rendering/ShowHide';
import Index from './prop-types';
import RouterSetup from './react-router';
import Form1 from './formik-react/Form1';
import FormikForm from './formik-react/Formik';

function App() {
  return (
    <div className="App">
        <FormikForm/>
    </div>
  );
}

export default App;
