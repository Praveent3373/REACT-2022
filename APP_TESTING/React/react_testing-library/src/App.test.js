import { cleanup, render } from '@testing-library/react';
import App from './App';

// Render a react component to the dom
// Use DOM APIs (query selector) to make assertions

afterEach(cleanup);

it('should take a snapshot', () => {
  const {asFragment} = render(<App/>)
  expect(asFragment(<App/>)).toMatchSnapshot();
});
