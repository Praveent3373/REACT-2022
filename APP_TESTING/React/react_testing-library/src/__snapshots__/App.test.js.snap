// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div>
      <h1
        data-testid="counter"
      >
        0
      </h1>
      <button
        data-testid="button-up"
      >
        increment 
      </button>
      <button
        data-testid="button-down"
      >
        decrement
      </button>
    </div>
  </div>
</DocumentFragment>
`;
